### PR TITLE
Wizard

### PR DESCRIPTION
**Pull request by RadekVana**
BFW-1269

### CODE REVIEW

File: src/gui/wizard/selftest_fans_axis.cpp [hunk 1]
```diff
@@ -1,12 +1,103 @@
-// selftest_fans_axis.c
+// selftest_fans_axis.cpp
+#include "selftest_fans_axis.hpp"
+#include "i18n.h"
+#include "gui.hpp"
+
+StateFncData StateFnc_SELFTEST_FAN0(StateFncData last_run) {
+    static const char *title_txt = N_(
```
Comment (DRracer) [inline src/gui/wizard/selftest_fans_axis.cpp:7]: Do we really need the extra space padding and explicit newlines?
Or is this piece of code subject to change in the further rework?

File: src/gui/wizard/xyzcalib.cpp [hunk 2]
```diff
@@ -1,15 +1,145 @@
...
+    static const string_view_utf8 title = _(
+        "Please remove steel "
+        "sheet from heatbed.");
+
+    MsgBox(title, Responses_NEXT);
+    return last_run.PassToNext();
+}
+
+StateFncData StateFnc_XYZCALIB_XY_MSG_PLACE_PAPER(StateFncData last_run) {
+    static const string_view_utf8 title = _(
```
Comment (DRracer) [inline src/gui/wizard/xyzcalib.cpp:72]: this cannot be static const, the current language may change during the runtime of the machine

File: src/gui/wizard/selftest_fans_axis.cpp (hunk 1 above)
Comment (RadekVana) [reply src/gui/wizard/selftest_fans_axis.cpp:7]: 1) I hope we do not. But do not know, did not try. 
It is just moved old code. 
2) Yes

Comment (RadekVana) [reply src/gui/wizard/selftest_fans_axis.cpp:7]: 1. Sadly we do
tested on
```
    static const char *title_txt = N_(
        "State\n"
        "SELFTEST_X"
        "not implemented");
```
\n is ignored

Comment (DRracer) [reply src/gui/wizard/selftest_fans_axis.cpp:7]: multiline flag?

File: src/gui/wizard/screen_wizard.cpp [hunk 3]
```diff
@@ -610,9 +610,8 @@ StateFncData StateFnc_FIRST(StateFncData last_run) {
 }
 
 StateFncData StateFnc_FINISH(StateFncData last_run) {
-    static const string_view_utf8 title = _(
-        "Calibration successful!\n"
-        "Happy printing!");
+    const char *txt = "Calibration successful!\nHappy printing!";
+    string_view_utf8 title = string_view_utf8::MakeRAM((const uint8_t *)txt);
```
Comment (DRracer) [inline src/gui/wizard/screen_wizard.cpp:614]: `static const char *txt = "Calibration successful!\nHappy printing!";
string_view_utf8 title = string_view_utf8::MakeCPUFLASH((const uint8_t *)txt);
`

File: src/gui/wizard/selftest.cpp [hunk 4]
```diff
@@ -15,31 +15,32 @@ StateFncData StateFnc_SELFTEST_INIT(StateFncData last_run) {
...
 StateFncData StateFnc_SELFTEST_FAIL(StateFncData last_run) {
-    static const string_view_utf8 title = _(
-        "The selftest failed\n"
-        "to finish.         \n"
-        "Double-check the   \n"
-        "printer's wiring   \n"
-        "and axes.          \n"
-        "Then restart       \n"
-        "the Selftest.      ");
+    const char *txt = "The selftest failed\n"
```
Comment (DRracer) [inline src/gui/wizard/selftest.cpp:25]: static const char * ... and tag it with N_()

**Review by XPila — DISMISSED** [review]

File: src/gui/wizard/selftest_fans_axis.cpp (hunk 1 above)
Comment (RadekVana) [reply src/gui/wizard/selftest_fans_axis.cpp:7]: Y

**Review by XPila — APPROVED** [review]

File: src/gui/wizard/selftest_fans_axis.cpp (hunk 1 above)
Comment (RadekVana) [reply src/gui/wizard/selftest_fans_axis.cpp]: did not help

**Pull request merged**